### DISCUSSION TITLE
How to get dnscrypt to 'resolve' onion sites

### DISCUSSION THREAD

**Discussion by Romulo-Romero** [DNSCrypt/dnscrypt-proxy#2633] [Q&A]
Hi everyone,
I've recently started using tor expert bundle without the tor browser, and I was wondering if there is anyway I can 'resolve' onion sites with dnscrypt, I've seen a post about forwarding rules, and I've enabled [onion   127.0.0.1:9053] config but it doesn't seem to make any difference, could anyone help me with this ?

Thanks in advance,
Romulo

**Comment by lifenjoiner** [comment]
Enable:
https://github.com/DNSCrypt/dnscrypt-proxy/blob/35d7aa0603182b73485ac371bc3b302735a4261c/dnscrypt-proxy/example-dnscrypt-proxy.toml#L381
and
https://github.com/DNSCrypt/dnscrypt-proxy/blob/35d7aa0603182b73485ac371bc3b302735a4261c/dnscrypt-proxy/example-forwarding-rules.txt#L36

Doesn't this work?

**Reply by Romulo-Romero** [reply]
I changed these two lines but it still doesn't work. I am running chromium browser and everything is being proxied through tor. I configured dnscrypt to listen on port 9050 and forced TCP, as a result I am able to browse through the Tor network.

**Comment by jedisct1** [comment]
You still need to run a Tor proxy.

**Reply by Romulo-Romero** [reply]
I am running the tor proxy, the only thing that I am not running is the Tor browser because I am proxying everything through the expert bundle proxy and using it with chromium browser.

**Reply by Romulo-Romero** [reply]
this is my Torrc file:
DataDirectory C:\xxx\xxx\Data\User
ClientOnionAuthDir C:\xxx\xxx\Data\User\onion-auth
GeoIPFile C:\xxx\xxx\Data\geoip
GeoIPv6File C:\xxx\xxx\Data\geoip6
Log notice stdout
CookieAuthentication 1
ControlPort 9051
SOCKSPort 127.0.0.1:9050
EntryNodes {fi},{no},{pl},{fr},{hr},{ch},{lu},{bg},{se},{lt},{de},{cz},{ro},{nl},{gb}
MiddleNodes {fi},{no},{pl},{fr},{hr},{md},{ch},{it},{lu},{bg},{se},{lt},{dk},{de},{cz},{ro},{nl},{hu},{gb},{at}
ExitNodes {no},{pl},{fr},{ch},{lu},{se},{de},{ro},{nl},{at}
HTTPTunnelPort 0

**Comment by Romulo-Romero** [comment]
I've added these two lines to my Torrc file:
DNSPort 127.0.0.1:9053
AutomapHostsOnResolve 1

So now my tor client is listening on port 9053 for dns.

![tor-listening](https://github.com/DNSCrypt/dnscrypt-proxy/assets/14086784/70ec43ff-2018-44e2-b7c3-cfdef40dd1c6)
![tor-log](https://github.com/DNSCrypt/dnscrypt-proxy/assets/14086784/ba106d08-1896-4b69-a237-9d5b8d5bcc28)

But I am still not able to reach onion sites, I've tried several onion sites from different web sites including the so called hidden wiki, but it still says error that it could not be resolved.

Duckduckgo: https://duckduckgogg42xjoc72x3sjasowoarfbgcmvfimaftt6twagswzczad.onion/

Tor status:
![tor-status](https://github.com/DNSCrypt/dnscrypt-proxy/assets/14086784/2bde22e1-7de2-418b-aca7-117208d4b303)

**Comment by lifenjoiner** [comment]
With all the above enabled, DNS queries on onion domains will work. But making other browsers work as a tor browser is another thing.
There are reasons why they customized a new browser. Tor browser is the easier and better option, and at last will be.
Study and try anything, explore it. That's the fun part.

**Reply by Romulo-Romero** [reply]
Alright, thanks for your input. I am going to keep trying things out and researching. I do not trust tor browser as much as my highly customized chromium, but that's ok, I am only going to use the tor browser for browsing the onion sites just out of curiosity and exploration.

> Study and try anything, explore it. That's the fun part.
I agree with you, so I will keep digging and see if I find interesting stuff

**Reply by anoduck** [reply]
The Domain Name System is a [standardized RFC specification](https://www.rfc-editor.org/rfc/rfc1035). 

Tor onion domains are a completely different specification, which is intentionally not standardized, because it will need to adapt and change over time. Not all onion addresses can be resolved 100% of the time. This is because Tor's domain system is decentralized and addresses are intentionally obfuscated from one node to the other. This is to ensure the protection of node identities within the network. 

Port 9053, is for forwarding standard DNS requests through the tor network. Tor onion domains will not be resolved on that port, because the tor network doesn't use that port for onion domain requests.

Even if you managed to peel away onion domain requests from tor, they would not be resolved into standard ipv4 or ipv6 addresses. So, the DNS server wouldn't recognize them anyway.
